### DISCUSSION TITLE
Writing Github secret to a file

### DISCUSSION THREAD

**Discussion by suvojit-0x55aa** [community/community#26973] [Actions]
<p>I’m trying to save secret to a file, it’s an important requirement for the build process.</p>
<pre><code>- name: Save Credentials
  run: 'echo "$SECRET_KEY" | base64 --decode &gt; secret_file.json'
  shell: bash
  env:
    SSH_KEY: ${{secrets.SECRET_KEY}}
</code></pre>
<p>But it’s only giving me empty file. How do I do it properly ?</p>

**Comment by jsoref** [comment, marked as answer]
<p>Your env variable was named  <strong>SSH_KEY</strong> not <strong>SECRET_KEY</strong> :</p>
<pre><code>- name: Save Credentials
  run: 'echo "$SSH_KEY" | base64 --decode &gt; secret_file.json'
  shell: bash
  env:
    SSH_KEY: ${{secrets.SECRET_KEY}}
</code></pre>

**Comment by suvojit-0x55aa** [comment]
<p>Oh okay, thanks. </p>
